src/templates/exportService.hbs

```diff
@@ -33,93 +36,115 @@ const {{{this}}}Validator = ajv.compile(${{{this}}});
-     * {{{description}}}
-    {{/if}}
-    {{#unless @root.useOptions}}
-    {{#if parameters}}
-    {{#each parameters}}
-     * @param {{{name}}} {{{description}}}
-    {{/each}}
-    {{/if}}
-    {{/unless}}
-    {{#each results}}
-     * @returns {{{type}}} {{{description}}}
-    {{/each}}
-     * @throws ApiError
-     */
-    public static async {{{name}}}({{>parameters}}): Promise<{{>result}}> {
-        const result = await __request({
-            method: '{{{method}}}',
-            path: `{{{path}}}`,
-            {{#if parametersCookie}}
-            cookies: {
-                {{#each parametersCookie}}
-                '{{{prop}}}': {{{name}}},
-                {{/each}}
-            },
-            {{/if}}
-            {{#if parametersHeader}}
-            headers: {
-                {{#each parametersHeader}}
-                '{{{prop}}}': {{{name}}},
-                {{/each}}
-            },
-            {{/if}}
-            {{#if parametersQuery}}
-            query: {
-                {{#each parametersQuery}}
-                '{{{prop}}}': {{{name}}},
-                {{/each}}
-            },
-            {{/if}}
-            {{#if parametersForm}}
-            formData: {
-                {{#each parametersForm}}
-                '{{{prop}}}': {{{name}}},
-                {{/each}}
-            },
-            {{/if}}
-            {{#if parametersBody}}
-            body: {{{parametersBody.name}}},
-            {{#if parametersBody.mediaType}}
-            mediaType: '{{{parametersBody.mediaType}}}',
-            {{/if}}
-            {{/if}}
-            {{#if responseHeader}}
-            responseHeader: '{{{responseHeader}}}',
-            {{/if}}
-            {{#if errors}}
-            errors: {
-                {{#each errors}}
-                {{{code}}}: `{{{description}}}`,
-                {{/each}}
-            },
-            {{/if}}
-            schemas: {
-                {{#each results}}
-                {{#unless isAny}}
-                {{#if isDefault}}
-                'default': {{{type}}}Validator,
-                {{else}}
-                '{{{code}}}': {{{type}}}Validator,
-                {{/if}}
-                {{/unless}}
-                {{/each}}
-            },
-        });
-        return result.body;
-    }
+interface IResult {
+    loading: Boolean;
+    error: CommonErrorResponseApplicationJsonResponse | null;
+    data: {{#if imports}}{{#each imports}}{{{this}}}|{{/each}}{{/if}}Number|String|null|undefined|Array<any>;
+}
 
-    {{/each}}
+interface IActionCallback {
+    (atom: IResult, set: (update: SetStateAction<IResult>) => void, result: ApiResult): void;
 }
+
+{{#each operations}}
+interface I{{{name}}}Controller {
+    fetchData: ({{>parameters}}
+        options?: ApiRequestOptions,
+        cb?: IActionCallback
+    ) => void;
+};
+
+{{/each}}
+
+{{#each operations}}
+export function {{{name}}}():[IResult, (update: SetStateAction<IResult>) => void, I{{{name}}}Controller] {
```

Comment:
1. Could the setter be placed inside the controller? It won't be used very often, maybe.
2. The function name should start with `use`, since it's a hook that called another hook.